.github/workflows/main.yml

```diff
@@ -16,6 +16,26 @@ on:
 
 jobs:
 
+  mingw:
+    runs-on: windows-latest
+
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        # Fetch all history to make sure tags are
+        # included (used for version creation)
+        fetch-depth: 0
+    - uses: seanmiddleditch/gha-setup-ninja@v3
+    - uses: Honeybunch/setup-mingw@v3
```

Comment:
I understand the reason for this. I have the same concern as for Ninja.

I am surprised that GitHub Actions is using an old version of MingW. It there perhaps a way to tell it to use a later version, e.g. `mingw-latest`? If not, you mentioned upstreaming your changes. Please do so.

Reply:
This is the PR I put up to upstream to the setup-mingw action I was going to use: https://github.com/egor-tensin/setup-mingw/pull/3

User egor-tensin is the only person I've found who has setup an action for setting up mingw. If my PR gets accepted I can go back and update this to use egor-tensin's version rather than mine. Either way the action packages themselves are hosted by Github so they're not going anywhere. 

A better PR for this might have been to modify the script to enable the action user to specify a version. The only complication I saw for that was that this action can also setup mingw on linux or through cygwin so it's interacting with a few different package managers.